Problem_Set_2/hangman.py

```diff
@@ -0,0 +1,400 @@
+
+# is_word_guessed('hello', ['h', 'r', 'e', 'l', 'k', 'm', 'o'])
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    guessed = ""
+    for char in secret_word:
+      if char in letters_guessed:
+        guessed += char + " "
+        # print(guessed)
+      else:
+        guessed += "_ "
+    return guessed
+    # pass
+
+# get_guessed_word('hello', ['i', 'r', 'j', 'b', 'k', 'm', 'o'])
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    letters = string.ascii_lowercase
+    letters_left = ""
+    for char in letters:
+      # print(char)
+      if char not in letters_guessed:
+        letters_left += char
+        # print(letters_left)
+    return letters_left
+    # print(letters_left)
+    # print(len(letters_left))
+    # pass
+    
+# get_available_letters(['i', 'r', 'j', 'b', 'k', 'm', 'o'])
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+    
+    Starts up an interactive game of Hangman.
+    
+    * At the start of the game, let the user know how many 
+      letters the secret_word contains and how many guesses s/he starts with.
+      
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    
+    * The user should receive feedback immediately after each guess 
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the 
+      partially guessed word so far.
+    
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    print("Welcome to the game Hangman!")
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    # print("Only enter letters. If you do not enter letters you will lose a warning and if you do not have any warnings left you will lose a guess")
+    print("You have 3 warnings left")
+    print("-------------")
+
+
+    warnings_left = 3
+    guesses_left = 6
+    letters_guessed = []
+    consonants = "bcdfghjklmnpqrstvwxyz"
+    vowels = "aeiou"
+
+    while guesses_left > 0 and is_word_guessed(secret_word, letters_guessed) == False:
```

Comment:
Is this code printing the word out?

Reply:
I don't get what you mean.

Reply:
The question said make sure to print out the guessed word after every round. The code doesn't seem to be doing that.

Reply:
it does. the guessed word is the one with the blanks and letters properly guessed. It prints it out every round. It's stored in the variable display, so i print display